### ISSUE TITLE
Faster app opening from tray

### ISSUE THREAD

**Issue by fsondej** [standardnotes/desktop#664]
I keep SN running and just minimize it to tray to open it faster. When I hit a shortcut that starts SN, it still takes about a second to open, even though it's running in the background. A second is not much, but I do it so often, that it feels quite sluggish.
It certainly can be done quicker, because when I right click on the tray icon and click 'show', it opens immediately.

The solution would probably require moving this if statement:
https://github.com/standardnotes/desktop/blob/afaddcf90ea48ece54ca6d11826f4587568ae00e/app/application.ts#L122-L126

at the beginning of initialization, before loading anything. I'm not sure though where is the 'focusing existing instance' part happening.

**Comment by JaspalSuri** [comment]
Hi @fsondej,

Sorry to hear that. Which operating system are you running Standard Notes on and what version of the desktop app is currently installed on your device? Are you using a passcode lock? Does your device use an SSD or mechanical hard drive?

**Comment by fsondej** [comment]
Operating System: Manjaro Linux
KDE Plasma Version: 5.20.5
Kernel Version: 5.9.16-1-MANJARO

Standard Notes v3.5.18

I'm using SSD and no passcode.

**Comment by baptiste-grob** [comment]
Hey @fsondej! 
The reason it takes this long is not because the `app.quit()` call happens too late (it actually happens very early on), but because when you open the app from a shortcut it starts off a new instance of electron, which is what is taking most of this 1-second delay and we don't have control over that, I'm afraid…

By the way, this is where the focusing happens:
https://github.com/standardnotes/desktop/blob/afaddcf90ea48ece54ca6d11826f4587568ae00e/app/application.ts#L78-L83

**Comment by fsondej** [comment]
Thanks for the quick reply!
Ah, I see, so we cannot focus without electron already started?

**Comment by baptiste-grob** [comment]
Indeed! We always need to start electron before doing anything. So unless they provide another mechanism for quick focus, we're unfortunately out of luck here.
